Base4/questions.js

```diff
@@ -1,101 +1,126 @@
-
+var removeSpaces = function(text) {
+  return text.trim();
 }
-var IsString = function (text) {
-
+var IsString = function(text) {
+  return (typeof(text) == "string");
 }
 
-var getExtension = function (text) {
-
+var getExtension = function(text) {
+  return text.slice(text.lastIndexOf(".") + 1, text.length);
 }
-var countSpaces = function (text) {
-
+var countSpaces = function(text) {
+  let counter = 0;
+  for (var i = 0; i < text.length; i++) {
+    if (text.charAt(i) == " ") counter++;
+  }
+  return counter;
 }
-var InverseString = function (text) {
+var InverseString = function(text) {
+  return text.split("").reverse().join("");
 
 }
 
-var power = function (x, y) {
-
+var power = function(x, y) {
+return Math.pow(x,y);
 }
-var absoluteValue = function (num) {
-
+var absoluteValue = function(num) {
+if (num<0)return -num;
+return num;
 }
-var absoluteValueArray = function (array) {
-
+var absoluteValueArray = function(array) {
+var absoluteArray = new Array();
+for(var i=0;i<array.length;i++){
+  if(array[i]<0){absoluteArray[i]=-array[i]}
+  else {absoluteArray[i]=array[i];}
 }
-var circleSurface = function (radius) {
-
+return absoluteArray;
 }
-var hypothenuse = function (ab, ac) {
-
+var circleSurface = function(radius) {
+return Math.round(3.14*radius*radius);
 }
-var BMI = function (weight, height) {
-
+var hypothenuse = function(ab, ac) {
+return Math.sqrt(ab*ab+ac*ac);
 }
-
-var createLanguagesArray = function () {
-
+var BMI = function(weight, height) {
+var bmi=weight/(height*height);
+return parseFloat(bmi.toFixed(2));
 }
 
-var createNumbersArray = function () {
-
+var createLanguagesArray = function() {
+return ['Html', 'CSS', 'Java', 'PHP'];
 }
 
-var replaceElement = function (languages) {
-
+var createNumbersArray = function() {
+return[0,1,2,3,4,5];
 }
 
-var addElement = function (languages) {
-
+var replaceElement = function(languages) {
+var arr=languages.slice(0);
+arr[2]="Javascript";
+return arr;
 }
 
-var addNumberElement = function (numbers) {
-
+var addElement = function(languages) {
+var arr=languages.slice(0);
+arr.push("Ruby");
+arr.push("Python");
+return arr;
 }
 
-var removeFirst = function (languages) {
-
+var addNumberElement = function(numbers) {
+var arr=numbers.slice(0);
+arr.unshift(-1);
+arr.unshift(-2);
+return arr;
 }
 
-var removeLast = function (languages) {
-
+var removeFirst = function(languages) {
+var arr= languages.slice(1);
+return arr;
 }
 
-var convertStrToArr = function (social_arr) {
-
+var removeLast = function(languages) {
+var arr= languages.slice(0,languages.length-1);
+return arr;
 }
 
-var convertArrToStr = function (languages) {
-
+var convertStrToArr = function(social_arr) {
+var arr=social_arr.split(",");
+return arr;
 }
 
-var sortArr = function (social_arr) {
-
+var convertArrToStr = function(languages) {
+var str=languages.join(",");
+return str;
 }
 
-var invertArr = function (social_arr){
+var sortArr = function(social_arr) {
+var arr=social_arr.sort();
```

Comment:
Be careful! `sort` sorts the array *in-place*. It does *not* return a new array.

To be consistent with everything else, you'd better `slice` the array before